r/R/scalar.R

```diff
@@ -33,7 +33,7 @@ Scalar <- R6Class("Scalar",
   public = list(
     ToString = function() Scalar__ToString(self),
     as_vector = function() Scalar__as_vector(self),
-    as_array = function() MakeArrayFromScalar(self),
+    as_array = function() MakeArrayFromScalar(self, 1L),
```

Comment:
If you pass this through, maybe you don't need `repeat_value_as_array`

```suggestion
    as_array = function(length = 1L) MakeArrayFromScalar(self, as.integer(length)),
```